### PR TITLE
Fix AcceptEncoding rule to support SOAP unit tests

### PR DESCRIPTION
**Pull request by rud2020**
Fix AcceptEncoding rule to support SOAP unit tests

### CODE REVIEW

File: lib/httpmock.js [hunk 1]
```diff
@@ -173,31 +208,13 @@ var server = http.createServer(function(request, response) {
...
-                        "Content-Encoding": "deflate"
-                    });
-                    zlib.deflate(matchingRule.response, (err, buffer) => {
-                        if (err) throw err;
-                        response.end(buffer);
-                    });
-                } else {
-                    response.end(matchingRule.response);
-                }
+                handleAcceptEncoding(
```
[the remark this answers is not in the document]
Comment (joaorosado) [reply lib/httpmock.js]: You should also add extra headers to your test just to make sure this scenario is covered.

File: lib/httpmock.js [hunk 2]
```diff
@@ -173,31 +208,13 @@ var server = http.createServer(function(request, response) {
...
+                    response,
+                    { ...request.headers, ...matchingRule.headers },
```
Comment (joaorosado) [inline lib/httpmock.js]: Don't get what we are doing here, why are we mixing the request headers with the response headers?

File: lib/httpmock.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (joaorosado) [reply lib/httpmock.js]: https://nodejs.org/api/http.html#http_response_writehead_statuscode_statusmessage_headers
"This method must only be called once on a message"

[the remark this answers is not in the document]
Comment (rud2020) [reply lib/httpmock.js]: Code Refactored, now all headers in matchedRule are included in the response

**Pull request merged**